### PR TITLE
Modernize user guide

### PR DESCRIPTION
**Pull request by jngrad**
Description of changes:
* use the *Alabaster* theme
   * responsive theme for mobile and tablets
   * visually light theme with consistent font sizes
   * natbib-style bibliography
* re-organize the user guide
   * collect information about running ESPResSo simulations into a self-contained chapter
   * move electrokinetics and reaction methods to self-contained chapters
   * make code examples more self-contained and remove pseudo-code
   * combine all BibTeX files into a single file
   * document running simulations in parallel with MPI
      * follow-up to mailing-list messages [Parallelization](https://lists.nongnu.org/archive/html/espressomd-users/2021-12/msg00011.html) and [max number of CPU's](https://lists.nongnu.org/archive/html/espressomd-users/2022-01/msg00001.html)

### CODE REVIEW

File: doc/bibliography.bib [hunk 1]
```diff
@@ -78,26 +143,73 @@ @Article{bindgen21a
...
+
 @ARTICLE{brodka04a,
   author = {Br\'{o}dka, A.},
-  title = {{E}wald summation method with electrostatic layer correction for interactions
-	of point dipoles in slab geometry},
-  journal = {Chem. Phys. Lett.},
+  title = {{E}wald summation method with electrostatic layer correction for interactions of point dipoles in slab geometry},
+  journal = {Chemical Physics Letters},
   year = {2004},
+  number = {1-3},
```
Comment (pkreissl) [inline doc/bibliography.bib]: This is also a number range, issues 1 to 3, so please use `1--3`.
The same goes for `crowl10a` (line 232)

File: doc/sphinx/ek.rst [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+``advection`` can be set to ``True`` or ``False``. It controls whether there should be an
+advective contribution to the diffusive species' fluxes. Default is
+``True``.
+
+``fluid_coupling`` can be set to ``"friction"`` or ``"estatics"``. This option determines the force
+term acting on the fluid. The former specifies the force term to be the
+sum of the species fluxes divided by their respective mobilities while
+the latter simply uses the electrostatic force density acting on all
+species. Note that this switching is only possible for the linkcentered
```
Comment (pkreissl) [inline doc/sphinx/ek.rst]: link-centered?

File: doc/sphinx/ek.rst [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+EK boundaries
+~~~~~~~~~~~~~
+::
+
+    ek_boundary = espressomd.ekboundaries.EKBoundary(charge_density=1.0, shape=my_shape)
+    system.ekboundaries.add(ek_boundary)
+
+.. note:: Feature ``EK_BOUNDARIES`` required
+
+The EKBoundary command allows one to set up (internal or external) boundaries for
```
Comment (pkreissl) [inline doc/sphinx/ek.rst]: ``EKBoundary``?

File: doc/sphinx/installation.rst [hunk 4]
```diff
@@ -33,39 +33,45 @@ software easily on a wide range of platforms.
...
-    files.
+    Boost
+        A number of advanced C++ features used by |es| are provided by Boost.
+        We strongly recommend to use at least Boost 1.71.
 
-MPI
-    Because |es| is parallelized with MPI, you need a working MPI
-    environment that implements the MPI standard version 1.2.
+    FFTW
+        For some algorithms like P\ :math:`^3`\ M, |es| needs the FFTW library
```
Comment (pkreissl) [inline doc/sphinx/installation.rst:52]: some algorithms [...] **need** (singular)

File: doc/sphinx/integration.rst [hunk 5]
```diff
@@ -0,0 +1,696 @@
...
+^^^^^^^^^^^^^^^^^^
+
+.. note::
+
+    Requires ``STOKESIAN_DYNAMICS`` external feature, enabled with
+    ``-DWITH_STOKESIAN_DYNAMICS=ON``.
+
+:meth:`espressomd.integrate.IntegratorHandle.set_stokesian_dynamics`
+
+The Stokesian Dynamics method allows to study the behaviour of spherical
```
Comment (pkreissl) [inline doc/sphinx/integration.rst]: I would go with AE "behavior" for consistency

File: doc/sphinx/io.rst [hunk 6]
```diff
@@ -93,32 +93,55 @@ An existing checkpoint can be loaded with::
...
 Be aware of the following limitations:
 
-* Checkpointing makes use of the ``pickle`` python package. Objects will only be restored as far as they support pickling. This is the case for Python's basic data types, ``numpy`` arrays and many other objects. Still, pickling support cannot be taken for granted.
+* Checkpointing makes use of the ``pickle`` python package. Objects will only
+  be restored as far as they support pickling. This is the case for Python's
+  basic data types, ``numpy`` arrays and many other objects. Still, pickling
+  support cannot be taken for granted.
 
-* Pickling support of the :class:`espressomd.system.System` instance and contained objects such as bonded and non-bonded interactions and electrostatics methods. However, there are many more combinations of active interactions and algorithms than can be tested.
+* Pickling support of the :class:`espressomd.system.System` instance and
```
Comment (pkreissl) [inline doc/sphinx/io.rst]: Obviously this is part of a item list. Still the meaning of this sentence is not immediately  clear to me (at least in the context of limitations) -- and grammatically there seems to be something missing.

File: doc/sphinx/io.rst [hunk 7]
```diff
@@ -93,32 +93,55 @@ An existing checkpoint can be loaded with::
...
+* The state of the cell system as well as the MPI node grid are checkpointed.
+  Therefore, checkpoints can only be loaded, when the script runs on the same
+  number of MPI ranks.
 
 * Checkpoints are not compatible between different |es| versions.
 
-* Checkpoints may depend on the presence of other Python modules at specific versions. It may therefore not be possible to load a checkpoint in a different environment than where it was loaded.
+* Checkpoints may depend on the presence of other Python modules at specific
+  versions. It may therefore not be possible to load a checkpoint in a
+  different environment than where it was loaded.
```
Comment (pkreissl) [inline doc/sphinx/io.rst]: than where it was **written**. (or saved?)

File: doc/sphinx/particles.rst [hunk 8]
```diff
@@ -140,59 +140,87 @@ To delete the exclusion::
...
+is controlled on a per-particle and per-axis level, where the axes
+are the Cartesian axes of the particle in its body-fixed frame.
+It is important to note that starting from version 4.0 and unlike
+in earlier versions of |es|, the particles' rotation is disabled by default.
 In this way, just compiling in the ``ROTATION`` feature no longer changes the physics of the system.
 
-The rotation of a particle is controlled via the :attr:`espressomd.particle_data.ParticleHandle.rotation` property. E.g., the following code adds a particle with rotation enabled around the x axis of its body frame::
+The rotation of a particle is controlled via the
+:attr:`~espressomd.particle_data.ParticleHandle.rotation` property.
+E.g., the following code adds a particle with rotation enabled around the x axis of its body frame::
```
Comment (pkreissl) [inline doc/sphinx/particles.rst]: x-axis (-- further down you use z-axis)

**Review by pkreissl — CHANGES_REQUESTED** [review]
In principle, your changes look good to me. For suggested changes see my comments.
Also, I did not build the doc yet, but I will do that later today to have a look at the visual and structure changes.

**Comment by jngrad** [comment]
> Also, I did not build the doc yet, but I will do that later today to have a look at the visual and structure changes.

To build the docs, first run `rm -r build/doc/sphinx`, otherwise sphinx will complain that there are non-included .rst files: these are old files that have been removed in the source directory, but CMake doesn't clean them up. Also make sure you have the latest Sphinx and sphinx-contrib versions. Alternatively, you can download the [zip artifact](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/314492/artifacts/download).

File: doc/sphinx/installation.rst (hunk 4 above)
Comment (jngrad) [reply doc/sphinx/installation.rst:52]: `|es|` expands to `ESPResSo`, which is singular

Comment (pkreissl) [reply doc/sphinx/installation.rst:52]: Sure, my bad, somehow I completely overlooked the `|es| ...

**Review by christophlohrmann — APPROVED** [review]
Thanks for the big effort.
I am a big fan of reducing the advanced methods section

**Review by pkreissl — APPROVED** [review]
Thanks, lgtm!

File: doc/sphinx/installation.rst [hunk 9]
```diff
@@ -814,172 +816,57 @@ parallel compilation. ``num_jobs`` specifies the maximal number of
...
-
-   Visual Studio Code interface
-
-.. _Debugging es:
+    cmake . -DCMAKE_BUILD_TYPE=Debug
+    make -j
 
-Debugging |es|
---------------
+The resulting build with be quite slow but will enable many debugging tools.
```
Comment (reinaual) [inline doc/sphinx/installation.rst:853]: typo

**Review by reinaual — APPROVED** [review]

File: doc/sphinx/installation.rst (hunk 9 above)
Comment (jngrad) [reply doc/sphinx/installation.rst:853]: good catch, I'll wrap this up when running the spellchecker during the release process next month

**Pull request merged**